### DISCUSSION TITLE
Why wait 1 second for internal consistency?

### DISCUSSION THREAD

**Discussion by dcecile** [brettcannon/gidgethub#182] [Q&A] [answered]
In the [quickstart example](https://github.com/brettcannon/gidgethub/blame/629b283e1986b7da5f57b45bc8d7c0d076cf9217/docs/index.rst#L72-L73), there's a 1 second sleep.

```py
# Give GitHub some time to reach internal consistency.
await asyncio.sleep(1)
```

What's the background behind this? Is it just for GitHub Apps and is it related to creating the `GitHubAPI` on the preceding line?

**Comment by brettcannon** [comment, marked as answer]
We have run into consistency issues in the past when performing a query against brand-new data. For instance, getting triggered on a new issue and then querying for that issue does not always lead to consistent/equivalent data in the query (e.g. the issue seemingly not existing). It was easiest to just sleep for one second and let GitHub catch up with itself. It's not a regular issue, but when things fail it was easier to just wait than toss the query out when something did go wrong.

**Reply by dcecile** [reply]
That makes sense, thanks 👍
